tests/test_log.cpp

```diff
@@ -70,3 +70,67 @@ TEST_F(log_tests, test_format_overload_multiple_args_missing)
     // time error with the C++20
     EXPECT_THROW({ mpl::log(mpl::Level::error, "test_category", "with formatting {} {}", 1); }, fmt::format_error);
 }
+
+// ------------------------------------------------------------------------------
+
+TEST_F(log_tests, test_log_error_function)
+{
+    logger_scope.mock_logger->expect_log(mpl::Level::error, "with formatting 1");
+    mpl::error("test_category", "with formatting {}", 1);
+}
+
+TEST_F(log_tests, test_log_warn_function)
+{
+    logger_scope.mock_logger->expect_log(mpl::Level::warning, "with formatting 1");
+    mpl::warn("test_category", "with formatting {}", 1);
+}
+
+TEST_F(log_tests, test_log_info_function)
+{
+    logger_scope.mock_logger->expect_log(mpl::Level::info, "with formatting 1");
+    mpl::info("test_category", "with formatting {}", 1);
+}
+
+TEST_F(log_tests, test_log_debug_function)
+{
+    logger_scope.mock_logger->expect_log(mpl::Level::debug, "with formatting 1");
+    mpl::debug("test_category", "with formatting {}", 1);
+}
+
+TEST_F(log_tests, test_log_trace_function)
+{
+    logger_scope.mock_logger->expect_log(mpl::Level::trace, "with formatting 1");
+    mpl::trace("test_category", "with formatting {}", 1);
+}
+
+// ------------------------------------------------------------------------------
+
+TEST_F(log_tests, test_log_error_function_noargs)
+{
+    logger_scope.mock_logger->expect_log(mpl::Level::error, "without formatting {}");
+    mpl::error("test_category", "without formatting {}");
```

Comment:
Hmm, should this be the expected output? So something like `mpl::error("category", "{} {}")` will log `"{} {}"` just fine, but `mpl::error("category", "{} {}", 1)` will give an error? Why not always treat the string as a format string?

Reply:
Because the user's intention might legitimately include printing "{}" without formatting (e.g., printing a JSON document as-is) [which would cause fmt::format to throw an error](https://gcc.godbolt.org/z/8G53enW4E). There are several instances of this in the codebase, and we need to be backward-compatible. 

If we're not formatting, we shouldn't experience the side effects of formatting either.

Reply:
ah, i see. for just printing {} with fmt::format, you could always use {{}}, but if this would break our existing code, it does make sense to have it like that. thanks for the explanation!

Reply:
I mean, these are new functions, and we can say that "these functions will always format, and you have to escape your braces", but that'd be inconsistent with the existing non-templated log overload.

```
mpl::error(category, "{}"); // runtime error, missing format argument
mpl::error(category, "{{}}"); // works fine, prints {}
mpl::log(Level::error, category, "{}"); // works fine, prints {}
```

Reply:
i feel like that would be more confusing. those functions should just introduce the log level, not have any extra special behavior IMO